moabb/paradigms/motor_imagery.py

```diff
@@ -82,43 +99,90 @@ def scoring(self):
+                                verbose=False, picks=picks)
+            xband, y = epochs.get_data(), epochs.events[:, -1]
+            X.append(xband)
+        return np.stack(X, axis=3), y
 
-    def __init__(self, n_classes, **kwargs):
-        self.n_classes = n_classes
-        super().__init__(**kwargs)
 
-    def verify(self, d):
-        print('Warning: Assumes events have already been selected per dataset')
-        super().verify(d)
-        assert len(d.selected_events) == self.n_classes
+def ImageryNClassFactory(parent):
 
+    class ImageryNClass(parent):
+        """Imagery for multi class classification
 
-class LeftRightImagery(BaseMotorImagery):
-    """Motor Imagery for left hand/right hand classification
+        Returns n-class imagery results, visualization agnostic but forces all
+        datasets to have exactly n classes. Uses 'accuracy' as metric
 
-    Metric is 'roc_auc'
+        """
 
-    """
+        def __init__(self, n_classes, **kwargs):
+            self.n_classes = n_classes
+            super().__init__(**kwargs)
 
-    def verify(self, d):
-        events = ['left_hand', 'right_hand']
-        super().verify(d)
-        assert set(events) <= set(d.event_id.keys())
-        d.selected_events = dict(zip(events, [d.event_id[s] for s in events]))
+        def verify(self, d):
+            log.warning(
+                'Assumes events have already been selected per dataset')
+            super().verify(d)
+            assert len(d.selected_events) == self.n_classes
 
-    @property
-    def scoring(self):
-        return 'roc_auc'
+        @property
+        def datasets(self):
+            return utils.dataset_search(paradigm='imagery',
+                                        total_classes=self.n_classes,
+                                        has_all_events=True)
 
-    @property
-    def datasets(self):
-        return utils.dataset_search(paradigm='imagery',
-                                    events=['right_hand', 'left_hand'],
-                                    has_all_events=True)
+    return ImageryNClass
+
+
+def LeftRightImageryFactory(parent):
+    class LeftRightImagery(parent):
+        """Motor Imagery for left hand/right hand classification
+
+        Metric is 'roc_auc'
+
+        """
+
+        def verify(self, d):
+            events = ['left_hand', 'right_hand']
+            super().verify(d)
+            assert set(events) <= set(d.event_id.keys())
+            d.selected_events = dict(
+                zip(events, [d.event_id[s] for s in events]))
+
+        @property
+        def scoring(self):
+            return 'roc_auc'
+
+        @property
+        def datasets(self):
+            return utils.dataset_search(paradigm='imagery',
+                                        events=['right_hand', 'left_hand'],
+                                        has_all_events=True)
+    return LeftRightImagery
+
+
+globals()['LeftRightImagerySinglePass'] = LeftRightImageryFactory(
```

Comment:
I wanted to add the classes to the namespace of moabb.paradigms.motor_imagery, and I think that would be the right way to do that? In the future there'll be other subfiles to moabb.paradigm and so i think this is a bit better for automatic code completion and such

I think the issue in the future is going to be that people will need to ensure that the pipeline files have the same paradigm names as are defined in moabb.paradigms somehow, and they'll need a simple way of linking the two -- right now if they make a spelling error in the pipeline or context .yml file it will throw an error while running, which isn't ideal.

Reply:
okay, I see :)

this is indeed an issue. i don't know how we can make this better for the user